### ISSUE TITLE
Data Import Version 10

### ISSUE THREAD

**Issue by ushashmi** [frappe/erpnext#12171]
Data Import tool seems to have some problem after its been upgraded to version 10. I tried following doc types

Supplier- on updating some fields in supplier doc.type its giving error that “naming_series is a mandatory field”. However in buying settings supplies are set by supplier names not by naming series.

User- upon any change in user data error message “argument of type ‘NoneType’ is not iterable”

**Comment by manassolanki** [comment]
@ushashmi Not able to replicate for the User import

**Comment by ushashmi** [comment]
@manassolanki I tried to disable few users and get that none type is not iterable

**Comment by jasonbrowne** [comment]
I am experiencing an issue importing items after fresh install of 10 for testing:
Frappe Framework
Close
Open Source Applications for the Web

	Website: https://frappe.io

	Source: https://github.com/frappe

Installed Apps

ERPNext: v10.0.2

Frappe Framework: v10.0.3

after selecting file to import system just stalls, refreshing the screen shows in progress which does not get updated.

**Comment by jasonbrowne** [comment]
![import-stalled-2017-12-26-14-47-47](https://user-images.githubusercontent.com/19347811/34366090-1f7aceea-ea4c-11e7-8f40-e1ae76f81f90.png)

**Comment by ushashmi** [comment]
@jasonbrowne are you updating the existing records or importing new items? 

In my test items import/update were fine

**Comment by Afetk** [comment]
Same issue as ushashmi. Try to export Supplier from erpnext (any variant of selection), then without changes try to upload - its giving error that “naming_series is a mandatory field”. Fix this by change in Buying Settings Supplier Naming By "naming_series" , then export date, add data to table, import, switch back to Settings Supplier Naming By "Supplier name"

**Comment by ushashmi** [comment]
@Afetk supplier issue is fixed already. Plz update to 10.02

**Comment by manassolanki** [comment]
Hi, @jasonbrowne Could you please comment the error log of that data import from the Error Log doctype.

**Comment by Afetk** [comment]
In Items table if export with data, than try to Update records have massage "Value cannot be changed for Attributes" (in case that any Attributes was not changed in file after import from system). It is heppend only if qwantity of attribute more than 2 (I have some Items with only 2 attribute - this line updates OK.

**Comment by tlie79** [comment]
Hi @manassolanki, I also just did Import on 27-12-2017 12:40:17 and had same failure. all 214 rows had error "argument of type 'NoneType' is not iterable". Trying to import Address DocType and having all the columns. ERPNext v10 latest.

**Comment by tlie79** [comment]
Hi @manassolanki, I notice there is the fix. When I do bench update, the import still fails. May I know how can I test this fix? thanks, Thomas

**Comment by manassolanki** [comment]
@tlie79 The fix hasn't been merged yet. You can check once it is merged.

**Comment by tlie79** [comment]
Thanks for your reply, @manassolanki. I copy your fix and merge manually in my local copy but it still failed. I'm trying to import Address Doctype. Would you be able to test on your side if the fix works for Address Doctype? Thanks.

**Comment by manassolanki** [comment]
@tlie79 Could you please comment the error log while importing the address.

**Comment by tlie79** [comment]
@manassolanki 

In the data import screen, each row has the following error:
argument of type 'NoneType' is not iterable
--

In the error log:
Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/core/doctype/data_import/importer.py", line 360, in upload
    doc, attachments, last_error_row_idx = get_doc(row_idx)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/core/doctype/data_import/importer.py", line 204, in validate_naming
    if autoname:
TypeError: argument of type 'NoneType' is not iterable

**Comment by manassolanki** [comment]
@tlie79 This has been fixed in the PR https://github.com/frappe/frappe/pull/4690

Please update your bench once we release. It will be merged into the master after release.

**Comment by tlie79** [comment]
@manassolanki 
I've actually copy & paste that fix into my local copy and still get the same error.
maybe I miss some steps, because this is the first time we try.

but anyway, we will wait for the offical release and try again. thanks.

**Comment by manassolanki** [comment]
It is released. You can check now

**Comment by tlie79** [comment]
Thank you for your prompt notification, @manassolanki. I have bench update & tried import again and it works successfully :+1: thanks for the fix.

out of curiousity, I follow the fix https://github.com/frappe/frappe/pull/4690 and copy into import.py. Why is it not working in my local? did I miss any steps?

**Comment by manassolanki** [comment]
Maybe there can be some tab and space related issue while copying the fix.
